src/streams/mod.rs

```diff
@@ -675,6 +672,11 @@ impl Streams {
         self.any_nack_active.unwrap()
     }
 
+    pub(crate) fn nack_at(&self) -> Option<Instant> {
+        // Return the earliest NACK time across all streams
```

Comment:
Why earliest and not latest? According to the logic of this code, it should return the time of the last requested NACK, not the first one. Or am I misunderstanding something?

Reply:
We return earliest because the poll loop needs to wake at the soonest deadline. If stream A needs NACK at 100ms and stream B at 200ms, we wake at 100ms to not miss A's deadline.